File: test/crc16.py
def crc16(x):
    b=0xA001
    a=0xFFFF
    for byte in x:
        a=a^ord(byte)
        for i in range(8):
            last=a%2
            a=a>>1
            if last==1:a=a^b
    aa='0'*(6-len(hex(a)))+hex(a)[2:]
    ll,hh=int(aa[:2],16),int(aa[2:],16)
    return hh+ll*256

print crc16('hello')
print crc16('world')

File: test/crc16.php
<?php 

function crc16($string) { 
    $crc = 0xFFFF; 
    for ($x = 0; $x < strlen ($string); $x++) { 
        $crc = $crc ^ ord($string[$x]); 
        for ($y = 0; $y < 8; $y++) { 
            if (($crc & 0x0001) == 0x0001) { 
                $crc = (($crc >> 1) ^ 0xA001); 
            } else { $crc = $crc >> 1; } 
        } 
    } 
    return $crc; 
} 


echo "crc16('hello')=" . crc16("hello") . "\n";
echo "crc16('world')=" . crc16("world") . "\n";
